ogr/ogrsf_frmts/miramon/mm_gdal_functions.c

```diff
@@ -0,0 +1,2801 @@
+{
+    int longitud_cadena = (int)l_cadena;
+    if (longitud_cadena-- == 0)
+        return punter;
+
+    if (punter[longitud_cadena] != ' ' && punter[longitud_cadena] != '\t')
+        return punter;
+    longitud_cadena--;
+
+    while (longitud_cadena > -1)
+    {
+        if (punter[longitud_cadena] != ' ' && punter[longitud_cadena] != '\t')
+        {
+            break;
+        }
+        longitud_cadena--;
+    }
+
+    punter[++longitud_cadena] = '\0';
+    return punter;
+}
+
+char *MM_RemoveInitial_and_FinalQuotationMarks(char *cadena)
+{
+    char *ptr1, *ptr2;
+    char cometa = '"';
+
+    if (*cadena == cometa)
+    {
+        ptr1 = cadena;
+        ptr2 = ptr1 + 1;
+        if (*ptr2)
+        {
+            while (*ptr2)
+            {
+                *ptr1 = *ptr2;
+                ptr1++;
+                ptr2++;
+            }
+            if (*ptr1 == cometa)
+                *(ptr1 - 1) = 0;
+            else
+                *ptr1 = 0;
+        }
+    }
+    return cadena;
+} /* End of MM_RemoveInitial_and_FinalQuotationMarks() */
+
+char *MM_RemoveLeadingWhitespaceOfString(char *cadena)
+{
+    char *ptr;
+    char *ptr2;
+
+    if (cadena == nullptr)
+        return cadena;
+
+    for (ptr = cadena; *ptr && (*ptr == ' ' || *ptr == '\t'); ptr++)
+        continue;
+
+    if (ptr != cadena)
+    {
+        ptr2 = cadena;
+        while (*ptr)
+        {
+            *ptr2 = *ptr;
+            ptr2++;
+            ptr++;
+        }
+        *ptr2 = 0;
+    }
+    return cadena;
+}
+
+char *MM_RemoveWhitespacesFromEndOfString(char *str)
+{
+    const char *s;
+
+    if (str == nullptr)
+        return str;
+
+    for (s = str; *s; ++s)
+        continue;
+    return MM_l_RemoveWhitespacesFromEndOfString(str, (s - str));
+}
+
+struct MM_ID_GRAFIC_MULTIPLE_RECORD *
+MMCreateExtendedDBFIndex(FILE_TYPE *f, MM_EXT_DBF_N_RECORDS nNumberOfRecords,
+                         MM_FIRST_RECORD_OFFSET_TYPE offset_1era,
+                         MM_ACUMULATED_BYTES_TYPE_DBF bytes_per_fitxa,
+                         MM_ACUMULATED_BYTES_TYPE_DBF bytes_acumulats_id_grafic,
+                         MM_BYTES_PER_FIELD_TYPE_DBF bytes_id_grafic,
+                         MM_BOOLEAN *isListField, MM_EXT_DBF_N_RECORDS *nMaxN)
+{
+    struct MM_ID_GRAFIC_MULTIPLE_RECORD *id;
+    MM_EXT_DBF_N_RECORDS i_dbf;
+    MM_EXT_DBF_SIGNED_N_RECORDS i, id_grafic;
+    char *fitxa;
+    MM_BYTES_PER_FIELD_TYPE_DBF bytes_final_id_principi_id1 =
+        bytes_per_fitxa - bytes_id_grafic;
+
+    *isListField = FALSE;
+    *nMaxN = 0;
+    if (!nNumberOfRecords)
+        return nullptr;  // No elements to read
+
+    if (MMCheckSize_t(nNumberOfRecords * sizeof(*id), 1))
+        return nullptr;
+    if (nullptr == (id = (struct MM_ID_GRAFIC_MULTIPLE_RECORD *)calloc_function(
+                        (size_t)nNumberOfRecords * sizeof(*id))))
+        return nullptr;
+
+    if (MMCheckSize_t(bytes_id_grafic + 1, 1))
```

Comment:
I'm not sure what this check will do given that bytes_id_graphic is of type MM_BYTES_PER_FIELD_TYPE_DBF == GUInt32. There can't be an overflow in the multiplication by 1. But what could theoretically happen if bytes_id_grafic is not validated is an overflow if bytes_id_graphic == UINT32_MAX